File: app/src/main/java/com/example/weatherapp1/Data/Model/Cloud.java
package com.example.weatherapp1.Data.Model;

import android.arch.persistence.room.ColumnInfo;
import android.os.Parcel;
import android.os.Parcelable;

public class Cloud extends ClassLoader implements Parcelable {
    @ColumnInfo(name = "cloud_type")
    private String type;

    public Cloud(String type) {
        this.type = type;
    }

    public String getType () {
        return type;
    }
    public void setAll (String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "Cloud{" +
                "type='" + type + '\'' +
                '}';
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.type);
    }

    protected Cloud(Parcel in) {
        this.type = in.readString();
    }

    public static final Parcelable.Creator<Cloud> CREATOR = new Parcelable.Creator<Cloud>() {
        @Override
        public Cloud createFromParcel(Parcel source) {
            return new Cloud(source);
        }

        @Override
        public Cloud[] newArray(int size) {
            return new Cloud[size];
        }
    };
}

File: app/src/main/java/com/example/weatherapp1/Data/Model/Cloud.kt
package com.example.weatherapp1.Data.Model

import android.arch.persistence.room.ColumnInfo
import android.os.Parcel
import android.os.Parcelable

class Cloud : ClassLoader, Parcelable {
    @ColumnInfo(name = "cloud_type")
    var type: String? = null
        private set

    constructor(type: String) {
        this.type = type
    }

    fun setAll(type: String) {
        this.type = type
    }

    override fun toString(): String {
        return "Cloud{" +
                "type='" + type + '\''.toString() +
                '}'.toString()
    }


    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.type)
    }

    protected constructor(`in`: Parcel) {
        this.type = `in`.readString()
    }

    companion object CREATOR : Parcelable.Creator<Cloud> {
        override fun createFromParcel(parcel: Parcel): Cloud {
            return Cloud(parcel)
        }

        override fun newArray(size: Int): Array<Cloud?> {
            return arrayOfNulls(size)
        }
    }
}
